scm_integrations.go

```diff
@@ -0,0 +1,120 @@
+package scalingo
+
+import (
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-scalingo/http"
+)
+
+type SCMType string
+
+// Type of SCM integrations
+const (
+	SCMGithubType           SCMType = "github"             // GitHub
+	SCMGithubEnterpriseType SCMType = "github-enterprise"  // GitHub Enterprise (private instance)
+	SCMGitlabType           SCMType = "gitlab"             // GitLab.com
+	SCMGitlabSelfHostedType SCMType = "gitlab-self-hosted" // GitLab self-hosted (private instance)
+)
+
+func (t SCMType) Str() string {
+	return string(t)
+}
+
+type SCMIntegrationsService interface {
+	SCMIntegrationsList() ([]SCMIntegration, error)
+	SCMIntegrationsShow(id string) (*SCMIntegration, error)
+	SCMIntegrationsCreate(scmType SCMType, url string, accessToken string) (*SCMIntegration, error)
+	SCMIntegrationsDelete(id string) error
+	SCMIntegrationsImportKeys(id string) ([]Key, error)
+}
+
+var _ SCMIntegrationsService = (*Client)(nil)
+
+type SCMIntegration struct {
+	ID          string  `json:"id"`
+	SCMType     SCMType `json:"scm_type"`
+	URL         string  `json:"url,omitempty"`
+	AccessToken string  `json:"access_token"`
```

Comment:
If you accepted the omitempty, on URL, shouldn't you add it here also?...

Reply:
The URL may well be empty, but access_token is still required